### PR TITLE
Add confirmation to project closeout flow

### LINKED ISSUE #685: PI needs to approve closeout
Closeout process should put project into new state pending PI approval.  Email goes out, then when PI approves the project is actually closed out via whatever process happened before.  Projects in closeout state probably shouldn't be able to have expenses entered and should only route to approval is the amount of money doesn't put the project over budget.

### PR DESCRIPTION
**Pull request by sprucely**
resolves #685

WIP. Still need to ensure expenses can't be added by other containers and in backend when project status is `PendingCloseoutApproval`. Oh, and forgot to check tests as usual.

### CODE REVIEW

**Comment by jSylvestre** [comment]
Looks good so far. I'll test again once the other stuff is integrated

**Review by srkirkland — COMMENTED** [review]
Looks good but I think you need to add some more history tracking to the action of initiating closeout.

File: Harvest.Web/ClientApp/src/Expenses/UnbilledExpensesContainer.tsx [hunk 1]
```diff
@@ -124,7 +125,10 @@ export const UnbilledExpensesContainer = (props: Props) => {
             )}
           </div>
           <div className="col text-right">
-            <ShowFor roles={["FieldManager", "Supervisor", "Worker"]}>
+            <ShowFor
+              roles={["FieldManager", "Supervisor", "Worker"]}
+              condition={project?.status !== "PendingCloseoutApproval"}
```
Comment (srkirkland) [inline Harvest.Web/ClientApp/src/Expenses/UnbilledExpensesContainer.tsx:130]: Why `project?.status`?  We've already checked to see that project !== undefined or else we wouldn't be showing any of this html at all

File: Harvest.Web/Controllers/Api/ExpenseController.cs [hunk 2]
```diff
@@ -33,16 +33,23 @@ public ExpenseController(AppDbContext dbContext, IUserService userService, IProj
...
-                expense.CreatedBy     = user;
-                expense.CreatedOn     = DateTime.UtcNow;
-                expense.ProjectId     = projectId;
-                expense.InvoiceId     = null;
-                expense.Account       = allRates.Single(a => a.Id == expense.RateId).Account;
+                expense.CreatedBy = user;
+                expense.CreatedOn = DateTime.UtcNow;
+                expense.ProjectId = projectId;
+                expense.InvoiceId = null;
+                expense.Account = allRates.Single(a => a.Id == expense.RateId).Account;
```
Comment (srkirkland) [inline Harvest.Web/Controllers/Api/ExpenseController.cs:52]: What do you have against tab alignment?

**Review by srkirkland — APPROVED** [review]

**Review by jSylvestre — APPROVED** [review]
Didn't test the call to sloth for the final invoice pending, but everything else looked good.

File: Harvest.Web/ClientApp/src/Expenses/UnbilledExpensesContainer.tsx (hunk 1 above)
Comment (sprucely) [reply Harvest.Web/ClientApp/src/Expenses/UnbilledExpensesContainer.tsx:130]: Didn't notice the `return <div>Loading...</div>;` part.

File: Harvest.Web/Controllers/Api/ExpenseController.cs (hunk 2 above)
Comment (sprucely) [reply Harvest.Web/Controllers/Api/ExpenseController.cs:52]: Nothing. Should I disable autoformat for c# code?

**Pull request merged**